### PR TITLE
fix(cli): prompted packages fail to install

### PR DESCRIPTION
**Pull request by AriPerkkio**
### Description

- Fixes https://github.com/vitest-dev/vitest/issues/4589

After updating `@antfu/install-pkg@0.2.0` the Rollup build fails to resolve `find-up > unicorn-magic`. There is `node` condition in its exports: https://github.com/sindresorhus/unicorn-magic/blob/67bb2fd1da1eecb252ba14284df14c0d234a8cfa/package.json#L16

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — CHANGES_REQUESTED** [review]
I feel like this is too broad. One of vitest entrypoints also runs in the browser - I am afraid this can introduce hard to debug issues. Maybe using `alias` plugin is safer in this case?

**Comment by AriPerkkio** [comment]
Agreed. I'll switch this to remap the `unicorn-magic`'s entrypoint with alias plugin.

File: packages/vitest/rollup.config.js [hunk 1]
```diff
@@ -77,9 +78,21 @@ const external = [
...
 const plugins = [
-  nodeResolve({
-    preferBuiltins: true,
+  alias({
+    entries: [
+      {
+        // unicorn-magic uses node conditions
+        // @antfu/install-pkg > find-up > unicorn-magic
+        find: 'unicorn-magic',
+        replacement: 'unicorn-magic',
```
[the remark this answers is not in the document]
Comment (sheremet-va) [reply packages/vitest/rollup.config.js:87]: This looks a bit weird to me, can't you just point to the file in replacement without resolver

[the remark this answers is not in the document]
Comment (AriPerkkio) [reply packages/vitest/rollup.config.js:87]: Something like these?

```js
{ find: 'unicorn-magic', replacement: 'unicorn-magic/node.js' },
{ find: 'unicorn-magic/default.js', replacement: 'unicorn-magic/node.js' }
```

The replacement should be an absolute path to the module. Not sure how to figure that out here. Is there a way to pass conditions for `require.resolve('unicorn-magic', { paths: [importer] })`?

[the remark this answers is not in the document]
Comment (AriPerkkio) [reply packages/vitest/rollup.config.js:87]: This works but now we are relying on `unicorn-magic` to build `node.js` file:

```js
find: 'unicorn-magic',
async customResolver(_, importer) {
  return resolve(dirname(require.resolve('unicorn-magic', { paths: [importer] })), 'node.js')
},
```

**Comment by AriPerkkio** [comment]
Updated to `0.3.0` but that still doesn't fix the issue where transitive dependency required `node` condition on module import.

```
[!] RollupError: "toPath" is not exported by "../../node_modules/.pnpm/unicorn-magic@0.1.0/node_modules/unicorn-magic/default.js", imported by "../../node_modules/.pnpm/find-up@7.0.0/node_modules/find-up/index.js".
https://rollupjs.org/troubleshooting/#error-name-is-not-exported-by-module
../../node_modules/.pnpm/find-up@7.0.0/node_modules/find-up/index.js (3:8)
1: import path from 'node:path';
2: import {locatePath, locatePathSync} from 'locate-path';
3: import {toPath} from 'unicorn-magic';
           ^
```

File: packages/vitest/rollup.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AriPerkkio) [reply packages/vitest/rollup.config.js:87]: > This looks a bit weird to me

This kind of usage is also documented on alias-plugin's docs: https://github.com/rollup/plugins/tree/master/packages/alias#custom-resolvers

We'll need the resolve plugin in order to automatically keep up with the `unicorn-magic`'s exports, without having to maintain hardcoded entries ourselves.

**Comment by antfu** [comment]
I guess it would be better to fix on `unicorn-magic`? I could also bundle `find-up` in `install-pkg` to workaround with this

**Comment by AriPerkkio** [comment]
`@antfu/install-pkg@0.3.1` works, thanks.

Verified fix with locally linked project and by manually checking `packages/vitest/dist/chunks/install-pkg.<hash>.js`.

**Comment by userquin** [comment]
@AriPerkkio we still have old version in root package.json

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**